File: AR7ExportToolbar/ToolStripExport/Exporters/RtfSectionReportExporter.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using GrapeCity.ActiveReports.Document;
using GrapeCity.ActiveReports.Export;

namespace AR7ExportToolbar.ToolStripExport.Exporters
{
	public class RtfSectionReportExporter : SectionReportExporter
	{
		private const string TypeName = "GrapeCity.ActiveReports.Export.Word.Section.RtfExport";
		private const string AssemblyName = "GrapeCity.ActiveReports.Export.Word.v7";

		public RtfSectionReportExporter()
			: base(TypeName, AssemblyName)
		{
		}

		public override void Export(SectionDocument document, string filename, NameValueCollection settings)
		{
			throw new NotImplementedException();
		}

		public override string DefaultExtension
		{
			get { return "rtf"; }
		}

		public override string FileDialogFilter
		{
			get { return "Rich-Text Files (*.rtf)|*.rtf|All Files (*.*)|*.*"; }
		}
	}
}

File: AR7ExportToolbarVB/ToolStripExport/Exporters/RtfSectionReportExporter.vb
Imports GrapeCity.ActiveReports.Document
Imports System.Collections.Specialized

Public Class RtfSectionReportExporter
	Inherits SectionReportExporter
	Private Const TypeName As String = "GrapeCity.ActiveReports.Export.Word.Section.RtfExport"
	Private Const AssemblyName As String = "GrapeCity.ActiveReports.Export.Word.v7"

	Public Sub New()
		MyBase.New(TypeName, AssemblyName)
	End Sub

	Public Overrides Sub Export(document As SectionDocument, filename As String, settings As NameValueCollection)
		Throw New NotImplementedException()
	End Sub

	Public Overrides ReadOnly Property DefaultExtension() As String
		Get
			Return "rtf"
		End Get
	End Property

	Public Overrides ReadOnly Property FileDialogFilter() As String
		Get
			Return "Rich-Text Files (*.rtf)|*.rtf|All Files (*.*)|*.*"
		End Get
	End Property
End Class
